prime-router/src/test/kotlin/azure/SettingsFunctionTests.kt

```diff
@@ -0,0 +1,581 @@
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+        response = settingsFunction.getOneOrganization(
+            httpRequestMessage,
+            "$otherOrganizationName.test-lab-2"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+    }
+
+    @Test
+    fun `test access user cannot update their organization's settings (updateOneOrganization)`() {
+        val settingsFunction = setupSettingsFunctionForTesting(oktaClaimsOrganizationName, mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting(HttpMethod.PUT)
+        val response = settingsFunction.updateOneOrganization(
+            httpRequestMessage,
+            "$organizationName.elr-secondary"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.UNAUTHORIZED)
+    }
+
+    @Test
+    fun `test access user cannot update another organization's settings (updateOneOrganization)`() {
+        val settingsFunction = setupSettingsFunctionForTesting(oktaClaimsOrganizationName, mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting(HttpMethod.PUT)
+        val response = settingsFunction.updateOneOrganization(
+            httpRequestMessage,
+            "$otherOrganizationName.test-lab-2"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.UNAUTHORIZED)
+    }
+
+    @Test
+    fun `test access DHPrimeAdmins can update any organization's settings (updateOneOrganization)`() {
+        val settingsFunction = setupSettingsFunctionForTesting("DHPrimeAdmins", mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting(HttpMethod.PUT)
+        var response = settingsFunction.updateOneOrganization(
+            httpRequestMessage,
+            "$organizationName.elr-secondary"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+        response = settingsFunction.updateOneOrganization(
+            httpRequestMessage,
+            "$otherOrganizationName.test-lab-2"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+    }
+
+    @Test
+    fun `test access user cannot get their organization's senders (getSenders-getOneSender)`() {
+        val settingsFunction = setupSettingsFunctionForTesting(oktaClaimsOrganizationName, mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting()
+        var response = settingsFunction.getSenders(
+            httpRequestMessage,
+            "$organizationName.elr-secondary"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.UNAUTHORIZED)
+        response = settingsFunction.getOneSender(
+            httpRequestMessage,
+            "$organizationName.elr-secondary",
+            "elr-secondary"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.UNAUTHORIZED)
+    }
+
+    @Test
+    fun `test access user cannot get another organization's senders (getSenders-getOneSender)`() {
```

Comment:
Assertion is correct here, but we should clean up the parameters and pass the organization, otherwise this isn't really asserting what we want.

Reply:
fixed